metis-authentication/metis-authentication-service/src/main/java/eu/europeana/metis/authentication/dao/ZohoAccessClientDao.java

```diff
@@ -141,8 +162,9 @@ private JsonNode findOrganizationFromListOfJsonNodes(JsonNode jsonOrgizationsRes
         JsonNode organizationField = organizationFields.next();
         JsonNode val = organizationField.get(VALUE_LABEL);
         JsonNode content = organizationField.get(CONTENT_LABEL);
-        if (val.textValue().equals(ORGANIZATION_NAME_FIELD) && content.textValue()
-            .equals(organizationName)) {
+        if (StringUtils.isNotEmpty(val.textValue()) && val.textValue()
+            .equals(ORGANIZATION_NAME_FIELD) && StringUtils.isNotEmpty(content.textValue())
+            && content.textValue().equals(organizationName)) {
```

Comment:
Maybe use StringUtils.equals as this also handles null values: you don't have to use StringUtils.isNotEmpty that way and there will only be two calls instead of four in this if statement.

Reply:
Indeed. Fixed